plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/errorsender/mq/KafkaErrorSenderPluginImpl.scala

```diff
@@ -0,0 +1,191 @@
+import org.apache.spark.sql.{DataFrame, Encoders}
+import za.co.absa.enceladus.plugins.api.postprocessor.PostProcessor
+import za.co.absa.enceladus.plugins.builtin.common.mq.kafka.KafkaConnectionParams
+import za.co.absa.enceladus.plugins.builtin.errorsender.DceError
+import za.co.absa.enceladus.plugins.builtin.errorsender.mq.KafkaErrorSenderPluginImpl.SingleErrorStardardized
+import za.co.absa.enceladus.utils.schema.SchemaUtils
+import KafkaErrorSenderPluginImpl._
+import za.co.absa.enceladus.plugins.builtin.errorsender.mq.kafka.KafkaErrorSenderPlugin
+import za.co.absa.enceladus.plugins.builtin.errorsender.params.ErrorSenderPluginParams
+import za.co.absa.enceladus.plugins.builtin.errorsender.params.ErrorSenderPluginParams.ErrorSourceId
+import za.co.absa.enceladus.utils.error.ErrorMessage.ErrorCodes
+
+import scala.util.{Failure, Success, Try}
+
+
+case class KafkaErrorSenderPluginImpl(connectionParams: KafkaConnectionParams,
+                                      keySchemaRegistryConfig: Map[String, String],
+                                      valueSchemaRegistryConfig: Map[String, String]) extends PostProcessor {
+
+  private val log = LogManager.getLogger(classOf[KafkaErrorSenderPluginImpl])
+
+  override def close(): Unit = {}
+
+  /**
+   * When data is ready, the error record(s) are pusblished to kafka.
+   *
+   * @param dataFrame error data only.
+   * @param params    Additional key/value parameters provided by Enceladus.
+   */
+  override def onDataReady(dataFrame: DataFrame, params: Map[String, String]): Unit = {
+    if (!SchemaUtils.fieldExists(ColumnNames.enceladusRecordId, dataFrame.schema)) {
+      throw new IllegalStateException(
+        s"${this.getClass.getName} requires ${ColumnNames.enceladusRecordId} column to be present in the dataframe!"
+      )
+    }
+
+    val errorSenderParams = Try(ErrorSenderPluginParams.fromMap(params)) match {
+      case Success(params) => params
+      case Failure(e) =>throw new IllegalArgumentException(
+        s"Incompatible parameter map supplied for ${KafkaErrorSenderPluginImpl.getClass.getName}: $params", e
+      )
+    }
+
+    val dfWithErrors = getIndividualErrors(dataFrame, errorSenderParams)
+
+    if (dfWithErrors.isEmpty) {
+      log.info("No errors to be sent to kafka.")
+    } else {
+      val count = dfWithErrors.count()
+      log.info(s"Sending $count errors to kafka topic ${connectionParams.topicName} ...")
+
+      Try {
+        val forKafkaDf = prepareDataForKafka(dfWithErrors, errorSenderParams)
+        sendErrorsToKafka(forKafkaDf)
+      } match {
+        case Success(_) => log.info(s"$count errors successfully sent to kafka topic ${connectionParams.topicName}")
+        case Failure(e) =>
+          log.error("Sending errors to kafka unsuccessful due to: ", e)
+      }
+    }
+  }
+
+  /**
+   * Processes the `dataFrame` - errors are exploded (one error = one line) and filtered to conform to the error source (standardization/conformance)
```

Comment:
backticks don't work in docs I think, you might want to use `{@code dataFrame}`

Reply:
Are you sure about this?
Everywhere I looked in the official scala doc, I saw
```
`backticked` 
or
{{{
codeblock
}}}
```
So, I am a little confused. I have confirmed that `{@code xxx}` works, too, to make stuff monospace, but I could not be sure that it is better for `"paramref"`

Reply:
Seems I am outdated. 👍